go.mod

```diff
@@ -2,12 +2,14 @@ module github.com/ionos-cloud/cluster-api-provider-proxmox
 
 go 1.20
 
+replace sigs.k8s.io/cluster-api => sigs.k8s.io/cluster-api v1.6.1
```

Comment:
do we actually want this?

Reply:
yes

Reply:
This will make sure to replace the nested dependencies of CAPI to always v1.6.1

Reply:
Please explain why this is needed

Reply:
There's other imported dependencies that have capi as dependency but different version, which leads to a conflict

Reply:
Then please add a TODO comment for removing the replace and name the problematic dependencies

Reply:
Note these circular dependencies are problematic for almost all providers.

https://github.com/kubernetes-sigs/cluster-api-provider-vsphere/blob/main/go.mod#L5

https://github.com/kubernetes-sigs/cluster-api-provider-gcp/blob/main/go.mod#L5

https://github.com/kubernetes-sigs/cluster-api-provider-aws/blob/main/go.mod#L8

https://github.com/kubernetes-sigs/cluster-api-provider-digitalocean/blob/main/go.mod#L132

https://github.com/kubernetes-sigs/cluster-api-provider-azure/blob/main/go.mod#L222

https://github.com/kubernetes-sigs/cluster-api-provider-ibmcloud/blob/main/go.mod#L9


I don't know how to document it and which deps are causing this circular dependency.

This is a way in go module to pin a dependency version.

Reply:
I agree. This should point at the version we're supporting.